### ISSUE TITLE
how to deal with the size of x and logits

### ISSUE THREAD

**Issue by 18972441546** [netw0rkf10w/CRF#6]
import torch

class CNNCRF(torch.nn.Module):
    """
    Simple CNN-CRF model
    """
    def __init__(self, cnn, crf):
        super().__init__()
        self.cnn = cnn
        self.crf = crf

    def forward(self, x):
        """
        x is a batch of input images
        """
        logits = self.cnn(x)
        logits = self.crf(x, logits)
        return logits

# Create a CNN-CRF model from given `cnn` and `crf`
# This is a PyTorch module that can be used in a usual way
model = CNNCRF(cnn, crf)

According to your usage, if my image input size is (1,3,512,512), the label shape is (1,2,512,512) where 2 is the category containing the background. The output logits shape after CNN processing is also (1,2,512,512). At this point, (1,3,512,512) and (1,2,512,512) will be input self.crf. I wonder if I can.

**Comment by netw0rkf10w** [comment]
> The output logits shape after CNN processing is also (1,2,512,512). At this point, (1,3,512,512) and (1,2,512,512) will be input self.crf. I wonder if I can.

Yes the input shapes should be good for `self.crf`.
